server/v2/server.go

```diff
@@ -28,18 +28,34 @@ type HasCLICommands interface {
 // HasConfig is a server module that has a config.
 type HasConfig interface {
 	Config() any
+	WriteConfig(string) error
```

Comment:
We should not need this. The server itself has a `WriteConfig` methods, that writes the all `app.toml`. Each server component shouldn't edit the `app.toml` manually.

Reply:
See https://github.com/cosmos/cosmos-sdk/blob/d7cc6de/server/v2/server.go#L130-L144

Reply:
resolved